vsphere/resource_vsphere_host_virtual_switch.go

```diff
@@ -0,0 +1,135 @@
+package vsphere
+
+import (
+	"fmt"
+	"time"
+
+	"context"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/govmomi"
+)
+
+func resourceVSphereHostVirtualSwitch() *schema.Resource {
+	s := map[string]*schema.Schema{
+		"name": &schema.Schema{
+			Type:        schema.TypeString,
+			Description: "The name of the virtual switch.",
+			Required:    true,
+			ForceNew:    true,
+		},
+		"host": &schema.Schema{
+			Type:        schema.TypeString,
+			Description: "The name of the host to put this virtual switch on. This is ignored if connecting directly to ESXi, but required if not.",
+			Optional:    true,
+			ForceNew:    true,
+		},
+		"datacenter": &schema.Schema{
+			Type:        schema.TypeString,
+			Description: "The path to the datacenter the host is located in. This is ignored if connecting directly to ESXi. If not specified on vCenter, the default datacenter is used.",
```

Comment:
This is tied to the [rationale for the host option](https://github.com/terraform-providers/terraform-provider-vsphere/pull/138#discussion_r135109717).